File: 0100-same-tree/0100-same-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   val: number
 *   left: TreeNode | null
 *   right: TreeNode | null
 *   constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 *   }
 * }
 */
function isSameTree(p: TreeNode | null, q: TreeNode | null): boolean {
  if (!p && !q) return true
  if (!p !== !q) return false
  if (!p.left && !p.right && !q.left && !q.right) return p.val === q.val
  return p.val !== q.val ? false : isSameTree(p.left, q.left) && isSameTree(p.right, q.right)
}

File: 0100-same-tree/0100-same-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *   Val int
 *   Left *TreeNode
 *   Right *TreeNode
 * }
 */
func isSameTree(p *TreeNode, q *TreeNode) bool {
  if p == nil && q == nil {
    return true
  }
  if (p == nil) != (q == nil) {
    return false
  }
  if (*p).Left == nil && (*p).Right == nil && (*q).Left == nil && (*q).Right == nil {
    return (*p).Val == (*q).Val
  }
  if (*p).Val != (*q).Val {
    return false
  } else {
    return isSameTree((*p).Left, (*q).Left) && isSameTree((*p).Right, (*q).Right)
  }
}